src/settings/Settings.tsx

```diff
@@ -367,6 +367,13 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         displayName: _td("New spotlight search experience"),
         default: false,
     },
+    "feature_jump_to_date": {
+        isFeature: true,
+        labsGroup: LabGroup.Messaging,
+        displayName: _td("Jump to date (adds /jumptodate). Also requires your homeserver to have MSC3030 enabled"),
+        supportedLevels: LEVELS_FEATURE,
+        default: false,
+    },
```

Comment:
This should probably check your server has MSC3030 enabled, like we do for hidden read receipt labs flag.

https://github.com/matrix-org/matrix-react-sdk/blob/b2548f05a825e3f2e38cbc6c04bfbe130105f7d4/src/components/views/settings/tabs/user/LabsUserSettingsTab.tsx#L59-L61

Reply:
Interesting! Looks like I need to update Synapse, https://github.com/matrix-org/synapse/blob/17886d2603112531d4eda459d312f84d0d677652/synapse/rest/client/versions.py#L78-L96

Reply:
It would also need adding to the MSC, e.g https://github.com/matrix-org/matrix-doc/pull/2432/files#diff-e29dd719b58cafbb5f6a02bb4bb17ac76e65476a58ea01a0c8b422efea6de03fR245-R247
Or you can just try hit the API with a random room and see if it responds with a 404.

Reply:
Thanks @t3chguy, I've created https://github.com/matrix-org/synapse/pull/11582 to add Synapse support ⏩

Updated the MSC (thanks for the great reference) https://github.com/matrix-org/matrix-doc/pull/3030

And now conditionally show the labs setting depending on homeserver support 🤘